### ISSUE TITLE
LDAP Sync broken

### ISSUE THREAD

**Issue by RousePartnersLLP-DB** [grokability/snipe-it#11425]
### Debug mode

- [x] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

Our LDAP sync does not work because the sync doesn't pull from sAMAccountName, it only gives a blank value, this causes Snipe to not inport any new accounts.

### Reproduction steps

1. Create new AD User
2. Sync AD user via LDAP
3. Read log of LDAP sync
...

### Expected behavior

Newly created AD users create in Snipe

### Screenshots

![image](https://user-images.githubusercontent.com/81681064/176468310-e3c40d5b-c4fa-484f-8218-4de8b624eb56.png)

User in AD ^

![image](https://user-images.githubusercontent.com/81681064/176468796-d7797341-bb05-4d36-856a-1957f06e556b.png)

User in Snipe LDAP Search ^

![image](https://user-images.githubusercontent.com/81681064/176469003-eaeff676-83bc-42e5-a4d3-52ffd725ba31.png)

sAMAccountName in AD

### Snipe-IT Version

Latest

### Operating System

Windows (AD Server)

### Web Server

Ubuntu

### PHP Version

?

### Operating System

Windows

### Browser

FireFox

### Version

100

**Comment by snipe** [comment]
We'd need to see what your LDAP settings are. Sync definitely works, it's likely just a misconfiguration.

**Comment by RousePartnersLLP-DB** [comment]
Hi there,

All settings are unchanged in recent weeks, and were working fine previously to latest version installation. Screenshots below, the three boxes above the first screenshot are all ticked:

**Comment by snipe** [comment]
Screenshots don't get attached when sending replies to GH issues via email. Please upload them via the web GUI.

**Comment by RousePartnersLLP-DB** [comment]
![image](https://user-images.githubusercontent.com/81681064/176471995-46280ec3-8134-4c59-bdbb-07a68496325f.png)

![image](https://user-images.githubusercontent.com/81681064/176472106-658a2fdc-0b0e-4e3e-8fd5-701c9aa9cfc9.png)

**Comment by snipe** [comment]
Per [the LDAP documentation](https://snipe-it.readme.io/docs/ldap-sync-login), try lowercasing the Username Field.

<img width="797" alt="Screen Shot 2022-06-29 at 8 13 37 AM" src="https://user-images.githubusercontent.com/197404/176472724-4be4f870-d7d4-440f-8109-6fdd1238a410.png">

<img width="799" alt="Screen Shot 2022-06-29 at 8 14 35 AM" src="https://user-images.githubusercontent.com/197404/176472731-628ed691-189d-4a8e-90ef-38535bcbc548.png">

**Comment by RousePartnersLLP-DB** [comment]
Why should that matter if it was working before?

**Comment by snipe** [comment]
Can you please just try it? If you recently upgraded, v6 is more strict about the filters - v5 basically ignored them.

**Comment by snipe** [comment]
(It has always been documented this way, since we originally wrote the LDAP integration back in v3. If this fixes your issue, your settings have been wrong the whole time.)

**Comment by RousePartnersLLP-DB** [comment]
> Can you please just try it? If you recently upgraded, v6 is more strict about the filters - v5 basically ignored them.
> (It has always been documented this way, since we originally wrote the LDAP integration back in v3. If this fixes your issue, your settings have been wrong the whole time.)

I really don't appreciate your tone here, in either comment. It's a completely valid question.

In this case I have tried it, and it does work. Frankly, all that this means is that your v5 implementation was superior. rather than our AD compliant methodology being "wrong". I really hope you do not talk to you paying support customers that way, it's a real put-off to us looking to move to your cloud instance w/ support.

**Comment by snipe** [comment]
> I really don't appreciate your tone here

Well, I don't appreciate people insisting that things are broken when they cannot be bothered to read the docs, so I guess we're even. I literally just asked you to *try it*, since it's come up a few times for users whose settings were configured incorrectly. There was no "tone".

> Frankly, all that this means is that your v5 implementation was superior. rather than our AD compliant methodology being "wrong". 

No, v5 was not superior. We had switched to using a library (instead of our handwritten implementation) that didn't support paging, so users with very large directories would never get a complete sync - which is obviously a huge problem.

In v6, we ripped out that library to accommodate larger directories, reverting back to v3-v4 code. We obviously wouldn't undo all of the work that went into v5's LDAP if not for very good reasons. The lower-case issue is a PHP LDAP thing, not something we're doing deliberately - which is why we're very sure to document it well. 

> I really hope you do not talk to you paying support customers that way, it's a real put-off to us looking to move to your cloud instance w/ support.

We're very nice to paying customers who get defensive when we're trying to help them, yes.
